pmd-core/src/main/java/net/sourceforge/pmd/lang/dfa/AbstractDataFlowNode.java

```diff
@@ -22,7 +22,7 @@
 
     protected List<DataFlowNode> parents = new ArrayList<>();
     protected List<DataFlowNode> children = new ArrayList<>();
-    protected BitSet type = new BitSet();
+    protected Set<NodeType> type = new HashSet<>();
```

Comment:
I'd change this to the more specific `EnumSet` type, since it's optimized for enums